lib/format/datetime/formatter.ex

```diff
@@ -94,10 +94,10 @@ defmodule Timex.Format.DateTime.Formatter do
     "#{hour}:#{minute}:#{sec}#{ms}"
   end
   def format_token(token, %DateTime{} = date, modifiers, _flags, _width)
-    when token in [:iso_8601, :iso_8601z] do
+    when token in [:iso_8601_extended, :iso_8601_extended_z] do
```

Comment:
Sadly, we need to keep the behaviour of the existing `{ISO}` and `{ISOz}` tokens (permissive parsing between extended and basic formats). If we implicitly treat them as `{ISO:Extended}` it may break assumptions people have. We'll have to keep the handling of the two sets of tokens separate, and deprecate `{ISO}` and `{ISOz}` over time.